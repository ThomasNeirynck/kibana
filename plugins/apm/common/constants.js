export const APP_NAME = 'context.app.name';
export const APP_AGENT_NAME = 'context.app.agent.name';
export const EVENT_PROCESSOR_NAME = 'processor.name';

export const TRANSACTION_DURATION = 'transaction.duration.us';
export const TRANSACTION_TYPE = 'transaction.type';
export const TRANSACTION_RESULT = 'transaction.result';
export const TRANSACTION_NAME = 'transaction.name';
export const TRANSACTION_ID = 'transaction.id';

export const TRACE_START = 'trace.start.us';
export const TRACE_DURATION = 'trace.duration.us';
export const TRACE_TRANSACTION_ID = 'trace.transaction_id';
export const TRACE_TYPE = 'trace.type';
export const TRACE_NAME = 'trace.name';
export const TRACE_ID = 'trace.id';
export const TRACE_SQL = 'context.db.statement';

export const ERROR_GROUP_ID = 'error.grouping_key';
export const ERROR_CULPRIT = 'error.culprit';
export const ERROR_MESSAGE = 'error.exception.message';
