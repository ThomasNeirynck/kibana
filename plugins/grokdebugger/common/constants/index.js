export { ROUTES } from './routes';
export { PLUGIN } from './plugin';
