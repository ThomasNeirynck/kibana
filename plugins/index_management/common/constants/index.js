export { PLUGIN } from './plugin';
export { BASE_PATH } from './base_path';
export * from './index_statuses';
