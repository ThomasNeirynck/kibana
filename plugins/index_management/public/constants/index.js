export { REFRESH_RATE_INDEX_LIST } from './refresh_intervals';
