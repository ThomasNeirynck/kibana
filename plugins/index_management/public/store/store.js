import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import { indexManagement } from './reducers/';

export const indexManagementStore = (initialState = {}) => {
  const enhancers = [ applyMiddleware(thunk) ];

  window.__REDUX_DEVTOOLS_EXTENSION__ && enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION__());
  return createStore(
    indexManagement,
    initialState,
    compose(...enhancers)
  );
};
