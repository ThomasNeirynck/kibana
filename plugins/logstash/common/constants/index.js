export { INDEX_NAMES } from './index_names';
export { TYPE_NAMES } from './type_names';
export { ROUTES } from './routes';
export { PAGINATION } from './pagination';
export { PLUGIN } from './plugin';
