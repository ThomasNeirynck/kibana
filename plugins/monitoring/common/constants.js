/**
 * The Monitoring API version is the expected API format that we export and expect to import.
 * @type {string}
 */
export const MONITORING_SYSTEM_API_VERSION = '2';
/**
 * The name of the Kibana System ID used to publish and lookup Kibana stats through the Monitoring system.
 * @type {string}
 */
export const KIBANA_SYSTEM_ID = 'kibana';
/**
 * The name of the Kibana System ID used to lookup Kibana stats through the Monitoring system.
 * @type {string}
 */
export const LOGSTASH_SYSTEM_ID = 'logstash';
/**
 * The type name used within the Monitoring index to publish Kibana stats.
 * @type {string}
 */
export const KIBANA_STATS_TYPE = 'kibana_stats';

/*
 * config options for welcome banner / allow phone home
 */
export const CONFIG_SHOW_BANNER = 'xPackMonitoring:showBanner';
export const CONFIG_ALLOW_REPORT = 'xPackMonitoring:allowReport';

/*
 * Chart colors
 */
export const CHART_LINE_COLOR = '#d2d2d2';
export const CHART_TEXT_COLOR = '#9c9c9c';
