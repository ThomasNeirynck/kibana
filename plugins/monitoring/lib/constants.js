/**
 * The Monitoring API version is the expected API format that we export and expect to import.
 * @type {string}
 */
export const MONITORING_SYSTEM_API_VERSION = '2';
/**
 * The name of the Kibana System ID used to publish Kibana stats through the Monitoring system.
 * @type {string}
 */
export const KIBANA_SYSTEM_ID = 'kibana';
/**
 * The type name used within the Monitoring index to publish Kibana stats.
 * @type {string}
 */
export const KIBANA_STATS_TYPE = 'kibana_stats';
