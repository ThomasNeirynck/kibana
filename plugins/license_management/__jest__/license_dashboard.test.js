import TestRenderer from 'react-test-renderer';
import React from 'react';
import Wrapper from '../public/app';
import { Provider } from 'react-redux';
import { licenseManagementStore } from "../public/store/store";
import { MemoryRouter } from 'react-router-dom';
import { BASE_PATH } from '../common/constants';

import sinon from 'sinon';

const highExpirationMillis = new Date("October 13, 2099 00:00:00Z");
const license = (type, expiryDateInMillis = highExpirationMillis) => {
  return  {
    type,
    expiryDateInMillis,
    isActive: true
  };
};
const getComponent = (initialState) => {
  const store = licenseManagementStore(initialState);
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={[ `${BASE_PATH}`]}>
        <Wrapper />
      </MemoryRouter>
    </Provider>
  );
};

describe('LicenseDashboard', () => {
  it('should display all panels for a trial license', () => {
    const rendered = TestRenderer.create(getComponent({ license: license('trial') }));
    expect(rendered).toMatchSnapshot();
  });
});
it('should display just license status for gold license', () => {
  const rendered = TestRenderer.create(getComponent({ license: license('gold') }));
  expect(rendered).toMatchSnapshot();
});
it('should display just license status and subscription for basic license', () => {
  const rendered = TestRenderer.create(getComponent({ license: license('basic') }));
  expect(rendered).toMatchSnapshot();
});
it('should display expiration notice for expired license', () => {
  const rendered = TestRenderer.create(getComponent({ license: license('gold', 0) }));
  expect(rendered).toMatchSnapshot();
});
it('should display basic registration pitch for license expiring in 25 days', () => {
  const clock = sinon.useFakeTimers(1506869972000);
  const rendered = TestRenderer.create(getComponent({ license: license('gold', 1508937678000) }));
  expect(rendered).toMatchSnapshot();
  clock.uninstall();
});

