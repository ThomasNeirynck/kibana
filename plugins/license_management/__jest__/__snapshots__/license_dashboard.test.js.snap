// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenseDashboard should display all panels for a trial license 1`] = `
<div
  className="kuiViewContent kuiViewContent--constrainedWidth"
>
  <div
    className="kuiNotice"
  >
    <div
      className="kuiInfoPanel kuiVerticalRhythm kuiInfoPanel--success"
    >
      <div
        className="kuiInfoPanelHeader__title kuiVerticalRhythmSmall"
      >
        <span
          aria-label="Check"
          className="kuiInfoPanelHeader__icon kuiIcon kuiIcon--success fa-check"
          role="img"
        />
        Your 
        trial
         license is 
        Active
      </div>
      <div
        className="kuiInfoPanelBody"
      >
        <div
          className="kuiInfoPanelBody__message"
        >
          Your license will expire on 
          <strong>
            October 12, 2099 7:00 PM EST
          </strong>
          .
        </div>
      </div>
    </div>
    <div
      className="kuiCardGroup kuiCardGroup--united kuiVerticalRhythm"
    >
      <div
        className="kuiCard"
      >
        <div
          className="kuiCard__description"
        >
          <div
            className="kuiCard__descriptionTitle"
          >
            Register for a Basic License
          </div>
          <div
            className="kuiCard__descriptionText"
          >
            Monitoring is available for free with a Basic License. Simply register, check your inbox, and enjoy!
          </div>
        </div>
        <div
          className="kuiCard__footer"
        >
          <a
            className="kuiButton kuiButton--secondary"
            href="https://register.elastic.co/xpack_register"
            target="_blank"
          >
            Get Basic
          </a>
        </div>
      </div>
      <div
        className="kuiCard"
      >
        <div
          className="kuiCard__description"
        >
          <div
            className="kuiCard__descriptionTitle"
            role="heading"
          >
            Unlock the Full Functionality
          </div>
          <div
            className="kuiCard__descriptionText"
          >
            Get all X-Pack has to offer: security, monitoring, alerting, reporting, and Graph, plus support from the engineers behind the Elastic Stack.
          </div>
        </div>
        <div
          className="kuiCard__footer"
        >
          <a
            className="kuiButton kuiButton--primary"
            href="https://www.elastic.co/subscriptions/xpack"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get It All
          </a>
        </div>
      </div>
    </div>
    <p
      className="kuiText kuiVerticalRhythm text-center"
    >
      Have a newer license? Splendid! 
      <a
        href="#/management/elasticsearch/license_management/upload_license"
      >
        Install it now.
      </a>
    </p>
  </div>
</div>
`;

exports[`should display basic registration pitch for license expiring in 25 days 1`] = `
<div
  className="kuiViewContent kuiViewContent--constrainedWidth"
>
  <div
    className="kuiNotice"
  >
    <div
      className="kuiInfoPanel kuiVerticalRhythm kuiInfoPanel--success"
    >
      <div
        className="kuiInfoPanelHeader__title kuiVerticalRhythmSmall"
      >
        <span
          aria-label="Check"
          className="kuiInfoPanelHeader__icon kuiIcon kuiIcon--success fa-check"
          role="img"
        />
        Your 
        gold
         license is 
        Active
      </div>
      <div
        className="kuiInfoPanelBody"
      >
        <div
          className="kuiInfoPanelBody__message"
        >
          Your license will expire on 
          <strong>
            October 25, 2017 9:21 AM EDT
          </strong>
          .
        </div>
      </div>
    </div>
    <div
      className="kuiCardGroup kuiCardGroup--united kuiVerticalRhythm"
    >
      <div
        className="kuiCard"
      >
        <div
          className="kuiCard__description"
        >
          <div
            className="kuiCard__descriptionTitle"
          >
            Register for a Basic License
          </div>
          <div
            className="kuiCard__descriptionText"
          >
            Monitoring is available for free with a Basic License. Simply register, check your inbox, and enjoy!
          </div>
        </div>
        <div
          className="kuiCard__footer"
        >
          <a
            className="kuiButton kuiButton--secondary"
            href="https://register.elastic.co/xpack_register"
            target="_blank"
          >
            Get Basic
          </a>
        </div>
      </div>
    </div>
    <p
      className="kuiText kuiVerticalRhythm text-center"
    >
      Have a newer license? Splendid! 
      <a
        href="#/management/elasticsearch/license_management/upload_license"
      >
        Install it now.
      </a>
    </p>
  </div>
</div>
`;

exports[`should display expiration notice for expired license 1`] = `
<div
  className="kuiViewContent kuiViewContent--constrainedWidth"
>
  <div
    className="kuiNotice"
  >
    <div
      className="kuiVerticalRhythm"
    >
      <h1
        className="kuiTitle kuiVerticalRhythmSmall"
      >
        <span
          className="kuiIcon fa-warning kuiIcon--error"
        />
        Your 
        gold
         license has expired
      </h1>
      <p
        className="kuiText kuiVerticalRhythmSmall"
      >
        Your license expired on 
        December 31, 1969 7:00 PM EST
        .
      </p>
    </div>
    <div
      className="kuiCardGroup kuiCardGroup--united kuiVerticalRhythm"
    >
      <div
        className="kuiCard"
      >
        <div
          className="kuiCard__description"
        >
          <div
            className="kuiCard__descriptionTitle"
          >
            Register for a Basic License
          </div>
          <div
            className="kuiCard__descriptionText"
          >
            Monitoring is available for free with a Basic License. Simply register, check your inbox, and enjoy!
          </div>
        </div>
        <div
          className="kuiCard__footer"
        >
          <a
            className="kuiButton kuiButton--secondary"
            href="https://register.elastic.co/xpack_register"
            target="_blank"
          >
            Get Basic
          </a>
        </div>
      </div>
      <div
        className="kuiCard"
      >
        <div
          className="kuiCard__description"
        >
          <div
            className="kuiCard__descriptionTitle"
            role="heading"
          >
            Unlock the Full Functionality
          </div>
          <div
            className="kuiCard__descriptionText"
          >
            Get all X-Pack has to offer: security, monitoring, alerting, reporting, and Graph, plus support from the engineers behind the Elastic Stack.
          </div>
        </div>
        <div
          className="kuiCard__footer"
        >
          <a
            className="kuiButton kuiButton--primary"
            href="https://www.elastic.co/subscriptions/xpack"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get It All
          </a>
        </div>
      </div>
    </div>
    <p
      className="kuiText kuiVerticalRhythm text-center"
    >
      Already have your license? Splendid! 
      <a
        href="#/management/elasticsearch/license_management/upload_license"
      >
        Install it now.
      </a>
    </p>
  </div>
</div>
`;

exports[`should display just license status and subscription for basic license 1`] = `
<div
  className="kuiViewContent kuiViewContent--constrainedWidth"
>
  <div
    className="kuiNotice"
  >
    <div
      className="kuiInfoPanel kuiVerticalRhythm kuiInfoPanel--success"
    >
      <div
        className="kuiInfoPanelHeader__title kuiVerticalRhythmSmall"
      >
        <span
          aria-label="Check"
          className="kuiInfoPanelHeader__icon kuiIcon kuiIcon--success fa-check"
          role="img"
        />
        Your 
        basic
         license is 
        Active
      </div>
      <div
        className="kuiInfoPanelBody"
      >
        <div
          className="kuiInfoPanelBody__message"
        >
          Your license will expire on 
          <strong>
            October 12, 2099 7:00 PM EST
          </strong>
          .
        </div>
      </div>
    </div>
    <div
      className="kuiCardGroup kuiCardGroup--united kuiVerticalRhythm"
    >
      <div
        className="kuiCard"
      >
        <div
          className="kuiCard__description"
        >
          <div
            className="kuiCard__descriptionTitle"
            role="heading"
          >
            Unlock the Full Functionality
          </div>
          <div
            className="kuiCard__descriptionText"
          >
            Get all X-Pack has to offer: security, monitoring, alerting, reporting, and Graph, plus support from the engineers behind the Elastic Stack.
          </div>
        </div>
        <div
          className="kuiCard__footer"
        >
          <a
            className="kuiButton kuiButton--primary"
            href="https://www.elastic.co/subscriptions/xpack"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get It All
          </a>
        </div>
      </div>
    </div>
    <p
      className="kuiText kuiVerticalRhythm text-center"
    >
      Have a newer license? Splendid! 
      <a
        href="#/management/elasticsearch/license_management/upload_license"
      >
        Install it now.
      </a>
    </p>
  </div>
</div>
`;

exports[`should display just license status for gold license 1`] = `
<div
  className="kuiViewContent kuiViewContent--constrainedWidth"
>
  <div
    className="kuiNotice"
  >
    <div
      className="kuiInfoPanel kuiVerticalRhythm kuiInfoPanel--success"
    >
      <div
        className="kuiInfoPanelHeader__title kuiVerticalRhythmSmall"
      >
        <span
          aria-label="Check"
          className="kuiInfoPanelHeader__icon kuiIcon kuiIcon--success fa-check"
          role="img"
        />
        Your 
        gold
         license is 
        Active
      </div>
      <div
        className="kuiInfoPanelBody"
      >
        <div
          className="kuiInfoPanelBody__message"
        >
          Your license will expire on 
          <strong>
            October 12, 2099 7:00 PM EST
          </strong>
          .
        </div>
      </div>
    </div>
    <div
      className="kuiCardGroup kuiCardGroup--united kuiVerticalRhythm"
    />
    <p
      className="kuiText kuiVerticalRhythm text-center"
    >
      Have a newer license? Splendid! 
      <a
        href="#/management/elasticsearch/license_management/upload_license"
      >
        Install it now.
      </a>
    </p>
  </div>
</div>
`;
